react-app/src/modules/account/account.test.js

```diff
@@ -0,0 +1,25 @@
+import * as account from './account.js';
+
+afterEach(() => account.clearAccount());
+
+test('creates keypairs from username and password', async () => {
+  expect.assertions(4);
+  await account.constructor('$/()RUFJ)wuwe84349', 'dnefu384(/·$873hf7g')
+    .then(result => {
+      expect(result).toBeNull();
+      expect(account.getUsername()).toBe('$/()RUFJ)wuwe84349');
+      expect(btoa(String.fromCharCode.apply(null, account.getPublicKey())))
```

Comment:
Having many different calls and values on the same line obscures what the code is doing, which makes harder to read the test, and most likely making easier to introduce bugs.
Adding comments is sometimes useful, but on this case I'd like you to use some auxiliary variables, we can use the variable names as documentation.

See this:

```js
  // this should be defined outside of the test and properly documented.
  const toBase64 = uint8 => btoa(String.fromCharCode.apply(null, uint8));

  const expectedPublicKey = "6FUygceUHlKzGu/5ir4FNkmAtUF07uNeAAdp9+5hcy4=";
  const publicKey = toBase64(account.getPublicKey());
  expect(publicKey).toEqual(expectedPublicKey);
```

It will make the test case longer, but much more readable as well.

Reply:
Couldn't you use `nacl.util.encodeBase64`?